tests/test_completion.py

```diff
@@ -158,37 +168,125 @@ def test_ignores_text_edit(self):
         self.assertEqual("$true", result[0])
         self.assertEqual("\\$true", result[1])
 
-    def test_item_with_partial_edit(self):
-        # issue #368, where using textEdit breaks PHP completions:
-        # $|
-        # PHP language server returns a partial completion "what"
-        # sublime uses it to replace '$' with 'what'
-        # to test:
-        # settings.complete_using_text_edit = True
+    def test_ignore_label(self):
+        # issue #368
         yield 100  # wait for file to be open
-
         item = {
-            'insertText': None,
-            # 'sortText': None,
-            # 'filterText': None,
-            'label': '$what',
+            'insertTextFormat': 2,
+            # insertText is present, but we should prefer textEdit instead.
+            'insertText': 'const',
+            'sortText': '3f800000const',
+            'kind': 14,
+            # Note the extra space here. We should ignore this!
+            'label': ' const',
+            'filterText': 'const',
             'textEdit': {
+                'newText': 'const',
                 'range': {
-                    'end': {'character': 1, 'line': 0},
-                    'start': {'character': 1, 'line': 0}
-                },
-                'newText': 'what'
+                    # Replace the single character that triggered the completion request.
+                    'end': {'character': 13, 'line': 6},
+                    'start': {'character': 12, 'line': 6}
+                }
             }
         }
-
         handler = CompletionHandler(self.view)
         handler.last_location = 1
         handler.last_prefix = ""
         result = handler.format_completion(item)
-        self.assertEqual(len(result), 2)
-        self.assertEqual("$what", result[0])
-        self.assertEqual("\\$what", result[1])
-        # settings.complete_using_text_edit = False
+        self.assertEqual(result, ('const\t  Keyword', 'const'))
+
+    def test_text_edit1(self):
```

Comment:
Would like to know the test name to describe difference between test_text_edit1 and test_text_edit2